tests/omprog-restart-terminated-outfile.sh

```diff
@@ -71,7 +71,7 @@ injectmsg 6 1
 injectmsg 7 1
 wait_queueempty
 
-kill -s USR1 $(get_child_pid)
+kill -s KILL $(get_child_pid)
```

Comment:
a word of caution: we are moving towards supporting parallel builds, as well es parallel execution of multiple CI runs on the same worker. I think the current `get_child_pid` implementation will not work in that environment. It would probably be good to change this to using child-created pidfiles some time in the future.

Reply:
If you recall, we already took that into consideration time ago in that test (and similar others). Please see some lines above:
```
# We need a test-specific program name, as the test needs to signal the child process
cp -f $srcdir/testsuites/omprog-restart-terminated-bin.sh $RSYSLOG_DYNNAME.omprog-restart-terminated-bin.sh

# On Solaris 10, the output of ps is truncated for long process names; use /usr/ucb/ps instead:
if [[ $(uname) = "SunOS" && $(uname -r) = "5.10" ]]; then
    function get_child_pid {
        echo $(/usr/ucb/ps -awwx | grep "$RSYSLOG_DYNNAME.[o]mprog-restart-terminated-bin.sh" | awk '{ print $1 }')
    }
else
    function get_child_pid {
        echo $(ps -ef | grep "$RSYSLOG_DYNNAME.[o]mprog-restart-terminated-bin.sh" | awk '{ print $2 }')
    }
fi
```
I.e. the program is renamed prior to spawning it by adding the test runtime random prefix. So in case of a parallel test execution, each test would have its own process name. (Note: because of the long process name, I had to use an alternate way to get the child PID in Solaris 10, as shown above.) Am I missing something?

> It would probably be good to change this to using child-created pidfiles some time in the future.
OK, seems a good idea since that would eliminate the need for the use of ps and also the Solaris special case. The pid file itself would also need to have a random name, though. Taking note of this as a future improvement.